django/db/migrations/operations/models.py

```diff
@@ -104,20 +104,25 @@ def describe(self):
         return "Create %smodel %s" % ("proxy " if self.options.get("proxy", False) else "", self.name)
 
     def references_model(self, name, app_label=None):
-        strings_to_check = [self.name]
+        name_lower = name.lower()
+        if name_lower == self.name_lower:
```

Comment:
It seems that this couldn't be done with the current approach.
I think that we could add `current_app` argument to `references_model` and `references_field` to handle it, but probable this should go to another PR.